pkg/processor/trigger/kafka/trigger.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2018 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafka
+
+import (
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/errors"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/processor/trigger"
+	"github.com/nuclio/nuclio/pkg/processor/worker"
+
+	"github.com/bsm/sarama-cluster"
+	"github.com/nuclio/logger"
+)
+
+type kafka struct {
+	trigger.AbstractTrigger
+	configuration  *Configuration
+	kafkaConfig    *cluster.Config
+	consumer       *cluster.Consumer
+	shutdownSignal chan struct{}
+}
+
+func newTrigger(parentLogger logger.Logger,
+	workerAllocator worker.Allocator,
+	configuration *Configuration) (trigger.Trigger, error) {
+	var err error
+
+	newTrigger := &kafka{
+		configuration: configuration,
+	}
+
+	newTrigger.AbstractTrigger = trigger.AbstractTrigger{
+		ID:              configuration.ID,
+		Logger:          parentLogger.GetChild(configuration.ID),
+		WorkerAllocator: workerAllocator,
+		Class:           "async",
+		Kind:            "kafka-cluster",
+	}
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create abstract stream")
+	}
+
+	newTrigger.Logger.DebugWith("Creating consumer", "url", configuration.URL)
+
+	newTrigger.kafkaConfig, err = newTrigger.newKafkaConfig(configuration.Attributes)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create configuration")
+	}
+
+	return newTrigger, nil
+}
+
+func (k *kafka) newKafkaConfig(attributes map[string]interface{}) (*cluster.Config, error) {
+
+	config := cluster.NewConfig()
+	config.Group.Mode = cluster.ConsumerModePartitions
+
+	return config, nil
+}
+
+func (k *kafka) newConsumer() (*cluster.Consumer, error) {
+	// init consumer
+	brokers := []string{k.configuration.URL}
+	consumer, err := cluster.NewConsumer(brokers, k.configuration.ConsumerGroup, k.configuration.Topics, k.kafkaConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create consumer")
+	}
+
+	k.Logger.DebugWith("Consumer created", "url", k.configuration.URL)
+	return consumer, nil
+}
+
+func (k *kafka) Start(checkpoint functionconfig.Checkpoint) error {
```

Comment:
Convention: 
Struct declarations
Public methods
Private methods

Please re-order the functions

Reply:
I'm going to assume there is no distinction between methods and ordinary functions.